distribution.pm

```diff
@@ -109,7 +110,12 @@ sub script_run {
     testapi::type_string "$cmd";
     if ($wait > 0) {
         my $str = testapi::hashed_string("SR$cmd$wait");
-        testapi::type_string " ; echo $str-\$?- > /dev/$testapi::serialdev\n";
+        if (testapi::is_serial_terminal) {
+            testapi::type_string " ; echo $str-\$?-\n";
+        }
+        else {
+            testapi::type_string " ; echo $str-\$?- > /dev/$testapi::serialdev\n";
+        }
```

Comment:
Let's focus on this refactoring in next PR, I would like to merge this PR this week.